cobra/core/Metabolite.py

```diff
@@ -40,6 +39,29 @@ def __init__(self, id=None, formula=None, name="",
         self._constraint_sense = 'E'
         self._bound = 0.
 
+    def _set_id_with_model(self, value):
+        if value in self.model.metabolites:
+            raise ValueError("The model already contains a metabolite with "
+                             "the id:", value)
+        self.model.solver.constraints[self.id].name = value
+        self._id = value
+        self.model.metabolites._generate_index()
+
+    @property
+    def constraint(self):
+        if self.model is not None:
+            return self.model.solver.constraints[self.id]
+
+    @property
+    def constraint(self):
```

Comment:
did you mean to repeat this three times? Also, could we get docstrings for new methods like this?